src/sentry/ingest/consumer_v2/factory.py

```diff
@@ -1,6 +1,6 @@
 from __future__ import annotations
```

Comment:
Agreed. @untitaker also suggested to create three complete separate strategies for the consumer types. I think we can explore this best in #50141, since I'd like to keep this PR to just tests and removal of old code.